packages/tezos-rust-libs/tezos-rust-libs.1.2/opam

```diff
@@ -0,0 +1,36 @@
+opam-version: "2.0"
+synopsis: "Tezos: all rust dependencies and their dependencies"
+maintainer: "contact@tezos.com"
+authors: "Tezos devteam"
+license: "multiple"
+homepage: "https://www.tezos.com/"
+bug-reports: "https://gitlab.com/tezos/tezos-rust-libs/issues"
+depends: ["conf-rust"]
```

Comment:
I think we need to add some constraints on the os/distributions/versions. Probably something like

```
available: [
  (os != "debian" | os-version > "10")
  | (os != "alpine" | os-version > "3.15")
  | …
]
```

I'm not entirely sure of the exact entry that needs to be added.
- Not sure what `os`, `os-family`, `os-distribution` combinations to use.
- Not entirely sure how the "testing" and "unsatble" versions of debians work, I assume there'll be a special case in the comparison function because that seems like something that would have been useful to other packages.
- etc.

I'd recommend you add those constraints locally based on the result of the CI, especially the "distributions" section. Then lint your opam files locally to make sure that it's parsable. And then push an amended version.

Reply:
Also, you can submit a `conf-rust-2021` package (see `conf-rust-2018` package for an example) and then you can depend on this versioned one instead. That might actually be a better place to set the os constraints!

Reply:
I added the `conf-rust-2021` package, but CI looks only marginally greener.

Reply:
I think the errors are better though, so that's already something.

Mostly, I think that the constraints on the `conf-rust-2021` need to be tighter. See other thread.